pkg/controller/machine-set-boot-image/platform_helpers.go

```diff
@@ -105,9 +105,58 @@ func checkMachineSet(infra *osconfigv1.Infrastructure, machineSet *machinev1beta
 	}
 }
 
-func reconcileAWS(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
-	klog.Infof("Skipping machineset %s, unsupported platform type AWS with %s arch", machineSet.Name, arch)
-	return false, nil, nil
+func reconcileAWS(machineSet *machinev1beta1.MachineSet, configMap *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
+	klog.Infof("Reconciling MAPI machineset %s on AWS, with arch %s", machineSet.Name, arch)
+
+	// First, unmarshal the AWS providerSpec
+	providerSpec := new(machinev1beta1.AWSMachineProviderConfig)
+	if err := unmarshalProviderSpec(machineSet, providerSpec); err != nil {
+		return false, nil, err
+	}
+
+	// Next, unmarshal the configmap into a stream object
+	streamData := new(stream.Stream)
+	if err := unmarshalStreamDataConfigMap(configMap, streamData); err != nil {
+		return false, nil, err
+	}
+	// Extract the region from the Placement field
+	region := providerSpec.Placement.Region
+	klog.Infof("AWS region: %s", region)
```

Comment:
nit: I think we can remove the info logs on this and line 133 below. If we feel that there is value in debugging, merging them is fine too

Reply:
Removed the info logs and moved the region to the this block 

	patchRequired = false
	newProviderSpec := providerSpec.DeepCopy()
	currentAMI := *newProviderSpec.AMI.ID
	if newami != currentAMI {
		klog.Infof("New target boot image: %s: %s", region, newami)
		klog.Infof("Current image: %s: %s", region, currentAMI)
		patchRequired = true
		newProviderSpec.AMI.ID = &newami
	}